### DISCUSSION TITLE
What is the best way to pass context between languages? (Java -> C++)

### DISCUSSION THREAD

**Discussion by ninomsu** [open-telemetry/opentelemetry-cpp#2254] [Q&A]
Currently I'm passing context from java to our c++ code as a string. The format, however, is not standard JSON and thus I don't have a great way of parsing it. Do I need to implement my own parser or is there a better way? I'd like to be able to reconstruct the message into a Span and use it for our c++ code.

What I'm receving on the the c++ side:

{opentelemetry-trace-span-key=SdkSpan{traceId=82af985588f983f1260e7b2f6d383fd1, spanId=1311808d606dedc4, parentSpanContext=ImmutableSpanContext{traceId=82af985588f983f1260e7b2f6d383fd1, spanId=a9be54d8115c4f63, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, name=send data, kind=SERVER, attributes=null, status=ImmutableStatusData{statusCode=UNSET, description=}, totalRecordedEvents=0, totalRecordedLinks=0, startEpochNanos=1690901592304165372, endEpochNanos=0}}

**Comment by lalitb** [comment]
There is no standard way designed to share the Context across languages. There have been some discussions here - https://github.com/open-telemetry/community/discussions/734. Most probably, in your case you need to write your own parser to deserialize the Span.
